tools/envoy_headersplit/headersplit_test.py

```diff
@@ -0,0 +1,107 @@
+# Lint as: python3
+"""Tests for headersplit."""
+
+import unittest
+import headersplit
+import io
+import clang.cindex
+from clang.cindex import TranslationUnit, Index, CursorKind
+clang.cindex.Config.set_library_path("/opt/llvm/lib")
+
+
+class HeadersplitTest(unittest.TestCase):
+  # A header contains a simple class print hello world
+  source_code_hello_world = open('code_corpus/hello.h', 'r').read()
+
+  # A C++ source code contains definition for several classes
+  source_class_defn = open('code_corpus/class_defn.h', 'r').read()
+
+  # almost the same as above, but classes are not enclosed by namespace
+  source_class_defn_without_namespace = open('code_corpus/class_defn_without_namespace.h',
+                                             'r').read()
+
+  # A C++ source code contains method implementaions for class_defn.h
+  source_class_impl = open('code_corpus/class_impl.cc', 'r').read()
+
+  def test_to_filename(self):
+    # Test class name with one "mock"
+    self.assertEqual(headersplit.to_filename("MockAdminStream"), "admin_stream")
+
+    # Test class name with two "Mock"
+    self.assertEqual(headersplit.to_filename("MockClusterMockPrioritySet"),
+                     "cluster_mock_priority_set")
+
+    # Test class name with no "Mock"
+    self.assertEqual(headersplit.to_filename("TestRetryHostPredicateFactory"),
+                     "test_retry_host_predicate_factory")
+
+  def test_get_directives(self):
+    includes = """// your first c++ program
+// NOLINT(namespace-envoy)
+#include <iostream>
+
+// random strings
+
+#include "foo/bar"
+
+"""
+
+    translation_unit_hello_world = TranslationUnit.from_source(
+        "code_corpus/hello.h", options=TranslationUnit.PARSE_SKIP_FUNCTION_BODIES)
+    self.assertEqual(headersplit.get_directives(translation_unit_hello_world), includes)
+
+  def test_class_definitions(self):
+    # class_definitions() should return a list of cursors, each pointing to one of those classes
```

Comment:
I think test cases should have Pydoc comments giving a quick intuition of what they do from a style perspective.